libraries/AP_InternalError/AP_InternalError.h

```diff
@@ -57,11 +61,17 @@ class AP_InternalError {
         switch_full_sector_recursion= (1U << 21),  //0x200000  2097152
         bad_rotation                = (1U << 22),  //0x400000  4194304
         stack_overflow              = (1U << 23),  //0x800000  8388608
+        __LAST__                    = (1U << 24),  // used only for sanity check
```

Comment:
Doesn't this have the slight technical downside that when you use the 32nd bit the `enum class` will be starting to use a 64 bit type everywhere?

Reply:
Do you think a comment would be sufficient to avoid that?  i.e. make those pesky future people worry about that.

Reply:
It's worth having for now, but it will be a problem in the future. You could also add a level of protection I think by just specifying the storage size of error_t to be uint32_t. Might be worth testing what happens once you try and overflow if the compiler will help you or not.

Reply:
I've added a comment and a static-assert.